tabpfn_client/prompt_agent.py

```diff
@@ -120,12 +137,29 @@ def prompt_and_set_token(cls, user_auth_handler: "UserAuthenticationClient"):
                 email = input(cls.indent("Please enter your email: "))
                 password = getpass.getpass(cls.indent("Please enter your password: "))
 
-                successful, message = user_auth_handler.set_token_by_login(
+                successful, message, status_code = user_auth_handler.set_token_by_login(
                     email, password
                 )
                 if successful:
                     break
-                print(cls.indent("Login failed: " + message) + "\n")
+                print(cls.indent("Login failed: " + str(message)) + "\n")
+                if status_code == 403:
+                    # Verify email
+                    verified = False
+                    while not verified:
```

Comment:
Somehow verification is duplicated? Why here and above?

Reply:
For registration, Login, and reverification. for login we need to set_token using email and password.
However, i have added a separate function now for verification functionality to avoid this - how about this?